api_fetch.js

```diff
@@ -1,131 +1,130 @@
-        container.append(descriptionElement);
-
-        // Create Book room button for each Challenge card
-        const button = document.createElement("button");
-        button.textContent = "Book this room";
-        button.dataset.challengeId = this.data.id;
-        container.append(button);
-
-        // Listen to button and forward challenge id to show title in booking modal
-        button.addEventListener("click", function (event) {
-            const challengeId = event.currentTarget.dataset.challengeId;
-            console.log("Challenge id:", challengeId);
-            document.getElementsByClassName("modal__stepOne")[0].style.display = "block";
-
-            const selectedChallenge = allChallenges.find(challenge => challenge.data.id === parseInt(challengeId));
-
-            const bookRoomTitle = document.createElement("h1");
-            bookRoomTitle.textContent = "Book Room: \"" + selectedChallenge.data.title + "\" (Step 1)";
-
-            const modalStepOne = document.querySelector("#challenge__title");
-            modalStepOne.append(bookRoomTitle);
-
-
-        });
-
-
-        return container;
-    }
+    // Create Book room button for each Challenge card
+    const button = document.createElement("button");
+    button.textContent = "Book this room";
+    button.dataset.challengeId = this.data.id;
+    container.append(button);
+
+    // Listen to button and forward challenge id to show title in booking modal
+    button.addEventListener("click", function (event) {
+      const challengeId = event.currentTarget.dataset.challengeId;
+      //document.getElementsByClassName("modal__stepOne")[0].style.display = "block";
+      displayModalStepOne(modal1);
+
+      selectedChallenge = allChallenges.find(
+        (challenge) => challenge.data.id === parseInt(challengeId)
+      );
+
+      const bookRoomTitle = document.querySelector("#modal1__title");
+      bookRoomTitle.textContent =
+        'Book Room: "' + selectedChallenge.data.title + '" (Step 1)';
+    });
+    return container;
+  }
 }
 
-
 // Request Challenge data from API and Create Challenge objects and put them in allChallenges array
 class APIadapter {
-    async getAllChallenges() {
-        const url = 'https://lernia-sjj-assignments.vercel.app/api/challenges';
-        const response = await fetch(url);
-        const payload = await response.json();
+  async getAllChallenges() {
+    const url = "https://lernia-sjj-assignments.vercel.app/api/challenges";
+    const response = await fetch(url);
+    const payload = await response.json();
 
-        allChallenges = payload.challenges.map((challengeData) => new Challenge(challengeData));
+    allChallenges = payload.challenges.map(
+      (challengeData) => new Challenge(challengeData)
+    );
 
-        return allChallenges;
-    }
+    return allChallenges;
+  }
 }
 
 //Global array to hold Challenge objects
-let allChallenges = [];
```

Comment:
You have moved the array, but you forgot to move the comment. This is a _very_ common mistake. My theory is that most programmers stop reading comments, unless they don't understand the code. The comments are colored differently and programmers, being lazy, just filter them out to have to read less. This makes sense, but it does mean that when you make changes to code, it's very easy to forget updating comments to reflect those changes.

For that reason, it's usually a good idea to avoid having comments, and instead, writing the code in a way that makes it easily readable, using good variable names etc. In this case, the comment ("Global array to hold Challenge objects") reveals little more than the variable name `allChallenges` and the fact that the variable is declared in the global scope, so it could probably have been omitted altogether.